prometheus/push/push.go

```diff
@@ -182,6 +185,14 @@ func (p *Pusher) BasicAuth(username, password string) *Pusher {
 	return p
 }
 
+// Format configures the Pusher to use an encoding format given by the
+// provided expfmt.Format. For convenience, this method returns a
+// pointer to the Pusher itself.
```

Comment:
Perhaps add what the default is (`expfmt.FmtProtoDelim`) and that that's the one most natural for the Prometheus Pushgateway, and that the Pushgateway will understand all the formats, but custom implementations of a push receiver might require a specific one.

Reply:
updated